### ISSUE TITLE
[macOS] Can’t add or remove displays

### ISSUE THREAD

**Issue by RustyMoyher** [libsdl-org/SDL#7764]
I discovered yesterday that after init, displays added or removed are ignored on macOS. `SDL_GetNumDisplayModes()` only provides the number at init. And I think new displays cannot be accessed.

Perhaps I'm missing something here? In SDL_cocoamodes.m (version 2.26.5):

- `SDL_AddVideoDisplay()` is only called on init
- `SDL_DelVideoDisplay()` is never called

I’m not sure of the correct way to start adding this. Maybe use `CGDisplayRegisterReconfigurationCallback` to detect the change?

**Comment by slouken** [comment]
That seems like a good place to start. @icculus, thoughts?

**Comment by slouken** [comment]
@RustyMoyher, did you try that? Do you have a PR for this?

**Comment by RustyMoyher** [comment]
No, I didn't make any progress on this, beyond just testing `CGDisplayRegisterReconfigurationCallback` to receive  notifications.


```
static void displayChanged(CGDirectDisplayID displayID, CGDisplayChangeSummaryFlags flags, void *userInfo)
{
    if (flags & kCGDisplayAddFlag) {
        // createDisplayConnectedEvent
    }
    if (flags & kCGDisplayRemoveFlag) {
        // createDisplayDisconnectedEvent
    }
}

...

CGDisplayRegisterReconfigurationCallback(displayChanged, NULL);
```

**Comment by slouken** [comment]
Okay, thanks!

**Comment by slouken** [comment]
@mikela-valve, relevant for your interests!

**Comment by icculus** [comment]
I believe Rusty is correct about the solution, and this has not been implemented yet for either SDL2 or SDL3.

**Comment by icculus** [comment]
I was delighted to discover that modern macOS can take AirPlay devices, like my Roku, and just treat them like normal displays, so I could test hotplugging by just clicking around in the system preferences and dragging windows to a TV across the room.  :)

**Comment by slouken** [comment]
Woo! :)
